server/functions/timestamptz.go

```diff
@@ -69,18 +72,18 @@ var timestamptz_out = framework.Function1{
 	Parameters: [1]*pgtypes.DoltgresType{pgtypes.TimestampTZ},
 	Strict:     true,
 	Callable: func(ctx *sql.Context, _ [2]*pgtypes.DoltgresType, val any) (any, error) {
+		if t, ok := val.(string); ok && (t == "infinity" || t == "-infinity") {
+			return t, nil
+		}
```

Comment:
Same comment here regarding `"Infinity"` string variations and lack of testing to catch this bug